src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String FRUIT = "fruit";
+    private static final String QUANTITY = "quantity";
+    private static final String SEPARATOR = ",";
+
+    @Override
+    public String generateReport() {
+        StringBuilder report = new StringBuilder();
+        report.append(FRUIT).append(SEPARATOR).append(QUANTITY).append(System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : Storage.storage.entrySet()) {

```

Comment:
The use of the Storage class directly ties your ReportGeneratorImpl to a specific storage implementation. Consider abstracting the storage access through a service or repository interface to follow the SOLID principles, making your ReportGeneratorImpl more independent and easier to test or modify in the future.